js/LocationButtons.js

```diff
@@ -0,0 +1,25 @@
+export default class LocationButtons extends HTMLElement {
+    constructor() {
+        super();
+        this.locationValue;
+        this.locationButtons = this.querySelectorAll('button');
+    }
+
+    connectedCallback() {
+        this.setupEventListener();   
+    }
+
+    setupEventListener() {
+        this.locationButtons.forEach( (button) => {
+            button.addEventListener('click', (ev) => {
+                this.locationValue = ev.target.value;
+                this.handleChange();
+            });
+        });
+    }
+
+    handleChange() {
+        const detail = this.locationValue;
+        this.dispatchEvent(new CustomEvent('changelocation', { detail, bubbles: true } ));
```

Comment:
Schön destructured mit `{ detail }`!

Reply:
(Und auch den Event hervorragend benannt)

Reply:
Es **kann** sich empfehlen, im `detail` von Anfang an gleich ein Objekt mitzugeben, z.B. `{ locationIdentifier: 'biel' }`, damit du später mal einfach erweitern kannst. Ist aber absolut in Ordnung, wie du's gemacht hast.